sycl/source/handler.cpp

```diff
@@ -2012,10 +2012,11 @@ void handler::setUserFacingNodeType(ext::oneapi::experimental::node_type Type) {
 }
 
 std::optional<std::array<size_t, 3>> handler::getMaxWorkGroups() {
-  auto Dev = detail::getSyclObjImpl(detail::getDeviceFromHandler(*this));
+  const auto &DeviceImpl =
+      MQueue ? MQueue->getDeviceImplPtr() : impl->MGraph->getDeviceImplPtr();
```

Comment:
Why not incapsulate this logic inside a function? The `getDeviceFromHandler` incapsulate this why not create the similar `getDeviceImplFromHandler`?

Reply:
But I found other places where the `getDeviceFromHandler` function is called.